### PR TITLE
Strongly check client isn't running on WebWorker for sendMsg

### LINKED ISSUE #745: Second argument of postMessage is not a transfert list
**Do you want to request a *feature* or report a *bug*?**

A bug.

**What is the current behavior?**

There is an error (only in webpack-dev-server, not in regular webpack builds) in the console when there is a target "webworker".

Chrome error:
> Uncaught TypeError: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope': The 2nd argument is neither an array, nor does it have indexed properties.

Firefox error:
> TypeError: Argument 2 of DedicatedWorkerGlobalScope.postMessage can't be converted to a sequence.

**If the current behavior is a bug, please provide the steps to reproduce.**

Testcase to reproduce it: https://github.com/wildpeaks/issue-webpack-dev-server-postmessage

They seem to point to [this line](https://github.com/webpack/webpack-dev-server/blob/v2.2.0-rc.0/client/index.js#L51) as it appears `"*"` is not a valid [transfert list](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage).

**What is the expected behavior?**

The expected behavior is no error shown in the console.

**Please mention your webpack and Operating System version.**

 - Node 7.4.0
 - Windows 10
 - Webpack 2.2.0-rc.7
 - Webpack-dev-server 2.2.0-rc.0

### PR DESCRIPTION
**Pull request by jhen0409**
**What kind of change does this PR introduce?** bugfix

**Summary**

Fix #745 again.

For some reason I've defined `self.window` in WebWorker, so #813 still doesn't fixed for me.

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function log(level, msg) {
 
 // Send messages to the outside, so plugins can consume it.
 function sendMsg(type, data) {
-	if(typeof self !== "undefined" && self.window) {
+	if(typeof self !== "undefined" && typeof WorkerGlobalScope === "undefined") {
```
Comment (shellscape) [inline client/index.js]: I'd like to amend this slightly:

```js
if(typeof self !== "undefined" && typeof WorkerGlobalScope === "undefined" && !self instanceOf WorkerGlobalScope) {
```

If we find another reason to check to see if the script is executing within a WebWorker, we'll want to move that out to a shared var.

Comment (jhen0409) [inline client/index.js]: I think the correct check should be this:

```js
typeof self !== "undefined" &&
(typeof WorkerGlobalScope === "undefined" ||
!(self instanceof self.WorkerGlobalScope))
```

Use `instanceof` only if it not undefined. I changed it. :)

File: client/index.js [hunk 2]
```diff
@@ -47,7 +47,11 @@ function log(level, msg) {
 
 // Send messages to the outside, so plugins can consume it.
 function sendMsg(type, data) {
-	if(typeof self !== "undefined" && self.window) {
+	if(
+		typeof self !== "undefined" &&
+		(typeof WorkerGlobalScope === "undefined" ||
+		!(self instanceof self.WorkerGlobalScope))
```
Comment (shellscape) [inline client/index.js]: I don't understand why you're using `self.WorkerGlobalScope` here. `self instanceof WorkerGlobalScope` should be correct.

Comment (jhen0409) [inline client/index.js]: Just passing eslint. Anyway I moved it to line 1.

**Pull request merged**